### PR TITLE
Share password policy

### PR DESCRIPTION
**Pull request by jotoeri**
So - should be good for a review now.
The ansatz on the vue file might not be perfect, but avoids esp. to have all the texts duplicated. I was still thinking about probably also inserting the renderShare by an initialState, so only having one asset. Might just need some kind of a loading view/feedback...

Corresponding changes on server to use the new policy are in  nextcloud/server#26491

Implements #117

### CODE REVIEW

File: appinfo/routes.php [hunk 1]
```diff
@@ -27,12 +27,28 @@
...
-			'url' => '/api/v1/validate',
+			'url' => '/api/{apiVersion}/validate',
+			'verb' => 'POST',
+			'requirements' => [
+				'apiVersion' => 'v(1|1\.1)',
+			],
+		],
+		[
+			'name' => 'API#generateShare',
+			'url' => '/api/v1.1/share/generate',
```
Comment (nickvergessen) [inline appinfo/routes.php:46]: all other apps just bump it as an integer atm, so would go with v2

Comment (jotoeri) [reply appinfo/routes.php]: Done 👍

**Comment by nickvergessen** [comment]
So we had some internal thoughts about this and are not sure if this is the right thing to do.
The problem is that there are pending tasks to add the password policy to other apps, e.g. talk for the room passwords, guest app user creation and so on. This would either mean that all those apps require another set of configuration values, or admins might not know to which of those passwords this one set of settings applies. Any idea for this?

cc @rullzer @LukasReschke

**Comment by jotoeri** [comment]
Hhm, so in general i'd interpret the current solution as two category-policies, one for all user-passwords and one for all sharing-passwords. As such, i'd e.g. apply the sharing-policy to the talk room-passwords, while guest-users fall below the user-policy. But i see your point, that it needs clarification of which one applies where.

Some ideas:
- Different Titles: For clarification i could change the Titles of the SettingsSection to 'User Password Poliy' and 'Share Password Policy'.
- How do you interpet the current sharing-settings page? Is this only thought for files-sharing? Then i'd need to shift the new policy to the security-page and apply different titles, so it becomes a general sharing-policy?
- Description: One could add a descriptive Text below the title like: 'Applies to: ...'. Indeed, this would need to be updated if a new app applies a policy.
- Using the info-icon of the settingssection one could also have a list of applicants on the docs, while just linking there from within the settings. Indeed also needs to be updated in case a new app includes a policy.
- Having the descriptive text ('Applies to') dynamically done with some config, so apps can register a text to get listed (interface towards apps, as e.g. the registration for activity-filters works in info.xml?). This might be possible, but a bit more complex? However, i could try to dig into, if this is the way to go 😉 


In general, i saw on our club-cloud already that admins strongly reduced (in fact disabled) the password-policy, as too many users complained about complex sharing-passwords. And i think having more places, where the policy is applied might even increase this problem. This just as an argument, why it might be good to have a second policy... ?🙈 😉

**Comment by jotoeri** [comment]
So - what is the state here? Is there a way to fix that? Any opinions on the proposed options? Or do you have other ideas?

**Pull request closed without merging**